llms/mlx_lm/tuner/datasets.py

```diff
@@ -83,12 +84,117 @@ def create_dataset(path: Path, tokenizer: PreTrainedTokenizer = None):
         )
 
 
+class TextHFDataset(Dataset):
+    """
+    A Huggingface dataset for a single blob of text per record
+    https://huggingface.co/docs/datasets/en/access
+    """
+
+    def __init__(self, hf_dataset, tokenizer, text_feature):
```

Comment:
You can make `text_feature` configurable in the regular dataset, default it to `text` and then just reuse that class.


Same comment for the completions dataset.